packages/api-client-core/spec/GadgetRecord.spec.ts

```diff
@@ -138,6 +138,23 @@ describe("GadgetRecord", () => {
     });
   });
 
+  it("should allow you to ask for CURRENT values of changed props on the entire object", () => {
+    const product = new GadgetRecord<SampleBaseRecord>(productBaseRecord);
+    expectNoChanges(product, "name", "body", "count");
+
+    product.name = "A newer name";
+    product.count = 123;
+
+    expect(product.changed("name")).toEqual(true);
+    expect(product.changed("count")).toEqual(true);
+    expect(product.changed("body")).toEqual(false);
+    expect(product.changed()).toEqual(true);
+    expect(product.currentChanges()).toEqual({
```

Comment:
I feel like the two functions `.currentChanges()` and `.changes()` don't exactly explain what is different about each other just looking at the names. What do you think about `.changes()` and `.changedJSON()` (or maybe `.toChangedJSON()`)?